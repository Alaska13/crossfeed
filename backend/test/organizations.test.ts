import * as request from 'supertest';
import { BACKEND_URL } from './constants';

describe('organizations', () => {
  it('add new organization', async () => {
    const name = 'cisa-test-' + Math.random();
    const response = await request(BACKEND_URL)
      .post('/organizations/')
      .send({
        ipBlocks: [],
        name,
        rootDomains: ['cisa.gov'],
        isPassive: false
      })
      .expect(200);
    expect(response.body).toMatchSnapshot({
      createdAt: expect.any(String),
      updatedAt: expect.any(String),
      id: expect.any(String),
      name: expect.any(String)
    });
    expect(response.body.name).toEqual(name);
  });
  it("can't add organization with the same name", async () => {
    const name = 'cisa-test-' + Math.random();
    await request(BACKEND_URL)
      .post('/organizations/')
      .send({
        ipBlocks: [],
        name,
        rootDomains: ['cisa.gov'],
        isPassive: false
      })
      .expect(200);
    const response = await request(BACKEND_URL)
      .post('/organizations/')
      .send({
        ipBlocks: [],
        name,
        rootDomains: ['cisa.gov'],
        isPassive: false
      })
      .expect(500);
    expect(response.body).toMatchSnapshot();
  });
});
