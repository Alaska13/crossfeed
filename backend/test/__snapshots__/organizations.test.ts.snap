// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organizations add new organization 1`] = `
Object {
  "createdAt": Any<String>,
  "id": Any<String>,
  "ipBlocks": Array [],
  "isPassive": false,
  "name": Any<String>,
  "rootDomains": Array [
    "cisa.gov",
  ],
  "updatedAt": Any<String>,
}
`;

exports[`organizations can't add organization with the same name 1`] = `""`;
